src/Pos/Update/Logic/Local.hs

```diff
@@ -89,6 +92,64 @@ getLocalVotes
     => m LocalVotes
 getLocalVotes = mpLocalVotes <$> getMemPool
 
+withCurrentTip
+    :: (Ether.MonadReader' UpdateContext m, MonadDB m)
+    => (MemState -> m MemState) -> m ()
```

Comment:
Is it OK that you don't return any confirmation flag saying was the state updated or not?